pkg/client/cache/controller.go

```diff
@@ -86,6 +86,10 @@ func New(c *Config) *Controller {
 // Run blocks; call via go.
 func (c *Controller) Run(stopCh <-chan struct{}) {
 	defer utilruntime.HandleCrash()
+	go func() {
```

Comment:
With defer, the function would never got invoked. Here it is important to reach the ``<-stopCh`` and wait.